src/input/json_to_storage.cc

```diff
@@ -371,15 +755,15 @@ StorageBase * JSONToStorage::make_storage(JSONPath &p, const Type::Record *recor
         		keys_to_process.erase(set_it);
         	}
 
-            if (p.down(it->key_) != NULL) {
+            if ( p->down(it->key_) ) {
                 // key on input => check & use it
                 storage_array->new_item(it->key_index, make_storage(p, it->type_.get()) );
-                p.up();
+                p->up();
             } else {
                 // key not on input
                 if (it->default_.is_obligatory() ) {
                     THROW( ExcInputError() << EI_Specification("Missing obligatory key '"+ it->key_ +"'.")
-                            << EI_ErrorAddress(p) << EI_InputType(record->desc()) );
+                            << EI_ErrorConstAddress(p) << EI_Format( p->input_format_name() ) << EI_InputType(record->desc()) );
```

Comment:
Probably better to catch the exception in read_storage and add EI_Format in order to reduce code duplicity.
That possibly may allow removing input_format_name.